r/R/dplyr-summarize.R

```diff
@@ -251,3 +268,21 @@ extract_aggregations <- function(expr, ctx) {
   }
   expr
 }
+
+# This function recurses through expr and wraps each call to median() with a
+# call to arrow_list_element()
+wrap_median <- function(expr, hash) {
+  if (length(expr) == 1) {
+    return(expr)
+  } else {
+    if (is.call(expr) && expr[[1]] == quote(median)) {
+      if (hash) {
+        return(str2lang(paste0("arrow_list_element(", deparse1(expr), ", 0L)")))
+      } else {
+        return(expr)
```

Comment:
I don't understand, you mean that hash_tdigest returns a list type but `tdigest` does not? That seems wrong, doesn't it?

Reply:
Apparently so; it doesn't work in the scalar aggregate case unless I do this 🤷‍♂️